packages/headless/src/features/result/result-insight-analytics-actions.ts

```diff
@@ -20,5 +21,29 @@ export const logDocumentOpen = (result: Result) =>
         documentIdentifier(result),
         metadata
       );
-    }
-  );
+    },
+    analyticsType: 'InsightPanel.ItemAction',
+    analyticsPayloadBuilder: (state): InsightPanel.ItemAction => {
+      const metadata = getCaseContextAnalyticsMetadata(
+        state.insightCaseContext
+      );
+      const identifier = documentIdentifier(result);
+      const information = partialDocumentInformation(result, state);
+      return {
+        itemMetadata: {
+          uniqueFieldName: identifier.contentIDKey,
+          uniqueFieldValue: identifier.contentIDValue,
+          title: information.documentTitle,
+          author: information.documentAuthor,
+          url: information.documentUri,
+        },
+        position: information.documentPosition,
+        searchUid: state.search?.response.searchUid || '',
+        action: 'open',
+        context: {
+          targetId: metadata.caseId || '',
+          targetType: 'Case',
+        },
```

Comment:
question do we not want to pass `caseNumber: metadata.caseNumber` in the context for logDocumentOpen ?

Reply:
Case number has been added!